### DISCUSSION TITLE
❓ Questions    Do you support realtime streaming input data?

### DISCUSSION THREAD

**Discussion by NathanJHLee** [snakers4/silero-vad#504] [Q&A] [answered]
## ❓ Questions and Help

We have a [wiki](https://github.com/snakers4/silero-models/wiki) available for our users. Please make sure you have checked it out first.

Hi, snakers4.
I built example to get vad result using c++.

Your system can control Realtime splited input data? 
I mean when
sample rate : 16000 
input wav : 10sec

totla input_data size = 160000 samples
split it every 1 second.
input size = 16000 * 10times 

std::vector<float> input(16000);
for (i=0; i<10; i++){
      vad.process(input); 
}
stamps = vad.get_speech_timestamps(); 

Then i couldn't get result.

Do you have any function to control realtime input data to get vad?
Thank you.

**Comment by snakers4** [comment, marked as answer]
Hi,

We have a real-time example in python - https://github.com/snakers4/silero-vad/blob/master/src/silero_vad/utils_vad.py#L383-L471

It is very naïve and it does not take into account all of the heuristics used in an offline example, but it's a good start.

Note that the VAD is not stateless.

**Comment by snakers4** [comment]
From the colab example

# Stream imitation example

## using VADIterator class

```python
vad_iterator = VADIterator(model, sampling_rate=SAMPLING_RATE)
wav = read_audio(f'en_example.wav', sampling_rate=SAMPLING_RATE)

window_size_samples = 512 if SAMPLING_RATE == 16000 else 256
for i in range(0, len(wav), window_size_samples):
    chunk = wav[i: i+ window_size_samples]
    if len(chunk) < window_size_samples:
      break
    speech_dict = vad_iterator(chunk, return_seconds=True)
    if speech_dict:
        print(speech_dict, end=' ')
vad_iterator.reset_states() # reset model states after each audio
```

## just probabilities

```python
wav = read_audio('en_example.wav', sampling_rate=SAMPLING_RATE)
speech_probs = []
window_size_samples = 512 if SAMPLING_RATE == 16000 else 256
for i in range(0, len(wav), window_size_samples):
    chunk = wav[i: i+ window_size_samples]
    if len(chunk) < window_size_samples:
      break
    speech_prob = model(chunk, SAMPLING_RATE).item()
    speech_probs.append(speech_prob)
vad_iterator.reset_states() # reset model states after each audio

print(speech_probs[:10]) # first 10 chunks predicts
```
